bin/cliain/src/snark_relations/relations.rs

```diff
@@ -273,8 +275,8 @@ impl GetPublicInput<CircuitField> for RelationArgs {
                 ..
             } => match (note, token_id, token_amount) {
                 (Some(note), Some(token_id), Some(token_amount)) => {
-                    DepositRelation::with_public_input(*note, *token_id, *token_amount)
-                        .public_input()
+                    DepositRelationWithPublicInput::new(*note, *token_id, *token_amount)
+                        .serialize_public_input()
```

Comment:
Could this be called simply _serialize_? We already have the `*WithPublicInput` in the type name.

Reply:
I'm slightly against - `serialize` on its own would indicate that we are serializing whole `RelationWithPublicInput` object, which might include constants and other irrelevant things

Reply:
OK, I'm convinced :)